airflow/jobs/local_task_job_runner.py

```diff
@@ -274,7 +275,14 @@ def heartbeat_callback(self, session: Session = NEW_SESSION) -> None:
             recorded_pid = ti.pid
             same_process = recorded_pid == current_pid
 
-            if recorded_pid is not None and (ti.run_as_user or self.task_runner.run_as_user):
+            is_child_process = (
+                ti.run_as_user
+                and (ti.run_as_user != getuser())
+                or self.task_runner.run_as_user
+                and (self.task_runner != getuser())
+            )
```

Comment:
Could you improve the parenthesis usage by respecting the pattern `(...) or/and (...)`?